go/vt/sqlparser/sql.y

```diff
@@ -4876,6 +5098,28 @@ alter_table_statement_part:
   {
     $$ = &DDL{Action: AlterStr}
   }
+| AUTO_INCREMENT equal_opt expression
+  {
+    $$ = &DDL{Action: AlterStr, AutoIncSpec: &AutoIncSpec{Value: $3}}
+  }
+| partition_operation
+  {
+    $$ = &DDL{Action: AlterStr, PartitionSpec: $1}
+  }
+
+with_or_without:
+  WITH
+  {
+    $$ = true
+  }
+| WITHOUT
+  {
+    $$ = false
+  }
+
+any_non_reserved:
+  ID
+| all_non_reserved
```

Comment:
Any chance `ID` can just go into `all_non_reserved`? It's pretty unclear from the names what the difference is between `any_non_reserved` and `all_non_reserved`. We have several rules for identifiers that have confusing names already, so would be nice to not add more if there's another way around it. If we really can't simplify then maybe there's a more intuitive name, such as `all_ids_and_non_reserved` or something that makes it clear what's different between that and `all_non_reserved`.